pkg/machine/wsl/wutil/wutil.go

```diff
@@ -114,13 +91,50 @@ func IsWSLStoreVersionInstalled() bool {
 	return true
 }
 
-func matchOutputLine(output io.ReadCloser, match string) bool {
-	scanner := bufio.NewScanner(transform.NewReader(output, unicode.UTF16(unicode.LittleEndian, unicode.UseBOM).NewDecoder()))
-	for scanner.Scan() {
-		line := scanner.Text()
-		if strings.Contains(line, match) {
-			return true
-		}
+func existsKernel() bool {
+	// from `MSI` or `Windows Update`
+	kernel := filepath.Join(getSystem32Root(), "lxss", "tools", "kernel")
+	if err := fileutils.Exists(kernel); err == nil {
+		return true
+	}
+
+	// from `Microsoft Store` or `Github`
+	kernel = filepath.Join(getProgramFiles(), "WSL", "tools", "kernel")
+	if err := fileutils.Exists(kernel); err == nil {
+		return true
 	}
+
 	return false
 }
+
+func getLocalAppData() (result string, ok bool) {
+	if local := os.Getenv("LOCALAPPDATA"); local != "" {
+		return local, true
+	}
+
+	if user := os.Getenv("USERPROFILE"); user != "" {
+		return filepath.Join(user, "AppData", "Local"), true
+	}
+
+	return "", false
+}
+
+func getSystem32Root() string {
+	if p := os.Getenv("SystemRoot"); p != "" {
+		return filepath.Join(p, "System32")
+	}
+
+	if p, err := windows.GetSystemDirectory(); err == nil {
+		return p
+	}
+
+	return `C:\Windows\System32`
+}
+
+func getProgramFiles() string {
```

Comment:
seems like some of these might be more usefully placed in general windows utils thing.  nit only, no worries

Reply:
Do you mean it's better to put: `getLocalAppData` `getSystem32Root` `getProgramFiles` in https://github.com/containers/podman/blob/main/pkg/machine/wsl/util_windows.go?

The reason I didn't put them in `util_windows.go` is because they belong to two different packages, and these three functions are only used here.
If you think it's necessary, I will move them to `util_windows.go`.